app.py

```diff
@@ -804,19 +805,38 @@ def tree_process(tree : ParentedTree, senttok: List[str]) -> ParentedTree:
 			subt.label = "-LRB-"
 		elif senttok[i] == "-RRB-":
 			subt.label = "-RRB-"
-		if (senttok[i] not in string.punctuation + "-LRB-" + "-RRB-") and subt.label in string.punctuation + "-LRB-" + "-RRB-":
+		if (senttok[i] not in string.punctuation + "-LRB-" + "-RRB-" + "...") and subt.label in string.punctuation + "-LRB-" + "-RRB-" + "...":
```

Comment:
I would make an `is_punct()` helper function that implements a definition of punctuation. I would not concatenate all these things together into one big string because then e.g. `"RB"` would count as punctuation. Instead, something like `re.match(r'^(\W+|-LRB-|-RRB-)$')` to test for punctuation.

Reply:
Note that `_.` will NOT count as punctuation per that regex because underscores are "word" characters.

Reply:
I don't think it is always possible to tell just from the token whether something is punctuation: "$", "&", "§" might be treated as lexical in certain contexts. So maybe `is_possible_punct()` is what's needed for checking compatibility of the lexical category.

Reply:
Great, this is addressed in https://github.com/nschneid/activedop/pull/47/commits/80285bf669d29dfb9550039f6940b06f10bce436. A variable `PUNCTRE` sets the regular expression for catching possible instances of punctuation. `AMBIG_SYM` defines a small set of symbols that could be either punctuation or non-punctuation. When a terminal matches `PUNCTRE`, the label is automatically changed to match the terminal (unless the terminal is in `AMBIG_SYM`, in which case the user has the option to treat it either as punctuation or non-punctuation).